lib/solidus_paypal_commerce_platform/engine.rb

```diff
@@ -15,10 +15,17 @@ class Engine < Rails::Engine
     engine_name 'solidus_paypal_commerce_platform'
 
     initializer "solidus_paypal_commerce_platform.add_payment_method", after: "spree.register.payment_methods" do |app|
-      app.config.spree.payment_methods << SolidusPaypalCommercePlatform::PaymentMethod
-      SolidusPaypalCommercePlatform::PaymentMethod.allowed_admin_form_preference_types << :paypal_select
-      Spree::PermittedAttributes.source_attributes.concat [:paypal_order_id, :authorization_id,
-                                                           :paypal_email, :paypal_funding_source]
+      app.config.to_prepare do
+        app.config.spree.payment_methods << SolidusPaypalCommercePlatform::PaymentMethod
+
+        unless SolidusPaypalCommercePlatform::PaymentMethod.allowed_admin_form_preference_types.include?(:paypal_select)
+          SolidusPaypalCommercePlatform::PaymentMethod.allowed_admin_form_preference_types << :paypal_select
+        end
```

Comment:
I made this push statement idempotent since the `to_prepare` block may run at least twice. See https://guides.rubyonrails.org/autoloading_and_reloading_constants.html#autoload-on-boot-and-on-each-reload.